### ISSUE TITLE
Eslint with Flow, Linter error with Class type

### ISSUE THREAD

**Issue by wojteg1337** [eslint/eslint#7945]
I'm using:
```json
{
    "eslint": "^3.13.1",
    "eslint-config-airbnb": "^14.0.0",
    "eslint-plugin-babel": "^4.0.0",
    "eslint-plugin-import": "^2.2.0",
    "eslint-plugin-jsx-a11y": "^3.0.2",
    "eslint-plugin-react": "^6.9.0",
    "flow-bin": "^0.36.0",
}
```

i create class:
```js
class LoginScreen extends Component<void, {}, LoginScreenState> {}
```

then when i try to create function:
```js
const createRoute = (screen: React.Element<*> | Class<*>, title: string) => ({ ... })
```

From this im getting 
> error no-undef: 'Class' is not defined.

**Comment by eslintbot** [comment]
Hi @wojteg1337, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by platinumazure** [comment]
Hi @wojteg1337, please provide all the items requested above, and please call out in particular which parser you're using to parse the Flow syntax. Thanks!

**Comment by nzakas** [comment]
No response, so closing.
